### PR TITLE
add resnet unit test

### PR DESCRIPTION
**Pull request by sharathns93**
Test that runs the same resnet graph on nGraph and CPU and compares the logits. 
Supports NHWC when using normal CPU and NCHW when built with MKL. 
Also supports different resnet sizes (--resnet_size) and training and inference phases (--phase_training=True)

### CODE REVIEW

**Comment by aprocter** [comment]
TESTNOW

**Comment by sharathns93** [comment]
Sounds good. Just let me know the location and I will move it there.

**Comment by sharathns93** [comment]
@avijit-nervana yes it is self-contained. will move it to `test/`

**Comment by sharathns93** [comment]
TESTNOW

**Comment by sharathns93** [comment]
TESTNOW

**Comment by aprocter** [comment]
Judging by the CI logs I think the script is getting caught up by the `pytest` run somehow. Maybe it needs to reside in a different directory since it is not a proper `pytest` test.
